### DISCUSSION TITLE
TS Config optimisations

### DISCUSSION THREAD

**Discussion by ariskemper** [open-circle/valibot#280] [Ideas]
@fabian-hiller why do you include all files in the main directory for TypeScript? 
We could include mod.ts, tsup.config.ts, vite.config.ts and exclude the rest like node_modules.
I would suggest also to set skipLibCheck: true to skip that TypeScript check declaration files.

Proposed TS Config Changes:
- include: ["mod.ts", "tsup.config.ts", "vite.config.ts"]
- exclude: ["node_modules"]
- skipLibCheck: true

**Comment by fabian-hiller** [comment]
What is the benefit of these changes? We do not use `tsconfig.json` in the build step and also use no dependencies beside Vitest.

**Reply by ariskemper** [reply]
@fabian-hiller it is for development purposes. Your IDE is using it, so at least include, exclude and skipLibCheck should be in tsconfig, otherwise your IDE is including also node_modules.

**Reply by fabian-hiller** [reply]
Ok. Do you want to create a PR or should I add it on my own?

**Reply by ariskemper** [reply]
@fabian-hiller will create a PR for that.
